contracts/libs/Authenticity.sol

```
@@ -32,6 +36,7 @@ library Authenticity {
                 ),
                 "Authenticity: state inclusion signer"
             );
+            require(proof.states[i].nonce > 0, "Authenticity: zero nonce");
```

Move this forward to fail fast